src/aa_remove_data/pb_utils.py

```diff
@@ -0,0 +1,164 @@
+        self.samples = []
+        self.sample_type = ""
+        if filepath:
+            self.read_pb(filepath)
+
+    def _escape_data(self, data: bytes) -> bytes:
+        """Replace escape characters with alternatives, to avoid conflitcs.
+        Should exactly mirror _unescape_data().
+
+        Args:
+            data (bytes): A serialised protobuf sample.
+
+        Returns:
+            bytes: The serialised sample with escape characters replaced.
+        """
+        data = data.replace(b"\x1b", b"\x1b\x01")  # Escape escape character
+        data = data.replace(b"\x0a", b"\x1b\x02")  # Escape newline
+        data = data.replace(b"\x0d", b"\x1b\x03")  # Escape carriage return
+        return data
+
+    def _unescape_data(self, data: bytes) -> bytes:
+        """Replace escape character alternatives with the escape characters.
+        Should exactly mirror _escape_data().
+
+        Args:
+            data (bytes): A serialised protobuf message with escape characters
+            replaced.
+
+        Returns:
+            bytes: The serialised protobuf message containing escape
+            characters.
+        """
+        data = data.replace(b"\x1b\x03", b"\x0d")  # Unescape carriage return
+        data = data.replace(b"\x1b\x02", b"\x0a")  # Unescape newline
+        data = data.replace(b"\x1b\x01", b"\x1b")  # Unescape escape character
+        return data
+
+    def _convert_to_class_name(self, sample_type: str) -> str:
+        """Convert the name of a sample type to CamelCase to correspond to
+        that sample type's class name.
+
+        Args:
+            sample_type (str): Name of sample type.
+
+        Returns:
+            str: Name of sample class, e.g VectorDouble.
+        """
+        # Split the enum name by underscores and capitalize each part
+        parts = sample_type.split("_")
+        return "".join(part.capitalize() for part in parts)
+
+    def convert_to_datetime(self, year: int, seconds: int) -> datetime:
+        """Get the date and time from a year and the number of seconds passed.
+        Args:
+            year (int): A year
+            seconds (int): The number of seconds into that year that have
+            passed.
+        Returns:
+            datetime: A datetime object of the correct date and time.
+        """
+        return datetime(year, 1, 1) + timedelta(seconds=seconds)
+
+    def get_datastr(self, sample: type, year: int) -> str:
+        """Get a string contaiing information about a sample.
+        Args:
+            sample (type): A sample from a pb file.
+            year (int): The year the sample was collected.
+        Returns:
+            str: A string containing the sample information.
+        """
+        date = self.convert_to_datetime(year, sample.secondsintoyear)
+        return (
+            f"{date}    {sample.secondsintoyear:8d}    {sample.nano:9d}"
+            f"    {sample.val}\n"
+        )
+
+    def get_sample_type(self) -> str:
```

Comment:
Thinking of some of the terminology here. Sample makes sense in a generic case, but perhaps it might be better to use terminology relating to EPICS and the Archiver Appliance specifically.
The `samples` in each PB file will be PVs. So perhaps this (and similar functions) should be called `get_pv_type` as that is the information we are obtaining from these files.

Reply:
This function is now called `get_pb_type`